src/stactools/landsat/stac.py

```diff
@@ -1,102 +1,281 @@
-
-    # Add thermal assets, if this is a L2SP product
-    if mtl_metadata.processing_level == 'L2SP':
-        for asset_definition in THERMAL_ASSET_DEFS:
+    sensor = Sensor(mtl_metadata.item_id[1])
+
+    if use_usgs_stac:
+        geometry = get_usgs_geometry(base_href, sensor,
+                                     mtl_metadata.product_id,
+                                     read_href_modifier)
+    else:
+        geometry = None
+
+    if geometry is None:
+        if sensor is Sensor.OLI_TIRS:
+            ang_href = ANG_ASSET_DEF.get_href(base_href)
+            ang_metadata = AngMetadata.from_file(ang_href, read_href_modifier)
+            geometry = ang_metadata.get_scene_geometry(mtl_metadata.bbox)
+        else:
+            geometry = mapping(box(*mtl_metadata.bbox))
+            logger.warning(
+                f"Using bbox for geometry for {mtl_metadata.product_id}.")
+
+    item = Item(id=mtl_metadata.item_id,
+                bbox=mtl_metadata.bbox,
+                geometry=geometry,
+                datetime=mtl_metadata.scene_datetime,
+                properties={})
+
+    if sensor is Sensor.OLI_TIRS:
+        item.common_metadata.platform = L8_PLATFORM
+        item.common_metadata.instruments = L8_INSTRUMENTS
+        item.common_metadata.description = L8_ITEM_DESCRIPTION
+
+        # eo
+        eo_item = EOExtension.ext(item, add_if_missing=True)
+        eo_item.cloud_cover = mtl_metadata.cloud_cover
+
+        # view
+        view = ViewExtension.ext(item, add_if_missing=True)
+        view.off_nadir = mtl_metadata.off_nadir
+        view.sun_elevation = mtl_metadata.sun_elevation
+        view.sun_azimuth = mtl_metadata.sun_azimuth
+
+        # projection
+        projection = ProjectionExtension.ext(item, add_if_missing=True)
+        projection.epsg = mtl_metadata.epsg
+        projection.bbox = mtl_metadata.proj_bbox
+
+        # landsat8
+        item.stac_extensions.append(L8_EXTENSION_SCHEMA)
+        item.properties.update(**mtl_metadata.landsat_metadata)
+
+        # -- Add assets
+
+        # Add common assets
+        for asset_definition in COMMON_ASSET_DEFS:
+            asset_definition.add_asset(item, mtl_metadata, base_href)
+
+        # Add SR assets
+        for asset_definition in SR_ASSET_DEFS:
             asset_definition.add_asset(item, mtl_metadata, base_href)
 
-    # -- Add links
+        # Add thermal assets, if this is a L2SP product
+        if mtl_metadata.processing_level == 'L2SP':
+            for asset_definition in THERMAL_ASSET_DEFS:
+                asset_definition.add_asset(item, mtl_metadata, base_href)
+
+        # -- Add links
+        # NOTE: This link is incorrect. Leads to a dead "page". The last
+        # component needs to have the processing time in it (replace item_id
+        # with product_id to fix. Needs to be resolved via a GitHub issue.
+        usgs_item_page = (
+            f"https://landsatlook.usgs.gov/stac-browser/collection02/level-2/standard/oli-tirs"
```

Comment:
Should probably be a constant (or partially a constant).

Reply:
Done.